src/composables/useApiGetMerchandise.ts

```diff
@@ -111,8 +126,205 @@ export const useApiGetMerchandise = (
+        .filter((variant) => variant.sex === item.sex)
+        .map((variant) => ({
+          label: variant.size,
+          value: variant.id,
+        }));
+      logger?.debug(`Size options <${JSON.stringify(sizeOptions, null, 2)}>`);
+
+      // create image array from t_shirt_preview
+      const images: Image[] = [
+        {
+          src: item.t_shirt_preview,
+          alt: item.name,
+        },
+      ];
+
+      // return MerchandiseItem
+      const merchandiseItem = {
+        id: item.id,
+        label: item.name,
+        gender: item.sex,
+        genderOptions,
+        size: item.size,
+        sizeOptions,
+        description: item.description,
+        images,
+      };
+      logger?.debug(
+        `Created item <${JSON.stringify(merchandiseItem, null, 2)}>`,
+      );
+      return merchandiseItem;
+    });
+
+    logger?.info(`Computed <${items.length}> merchandise items`);
+    return items;
+  };
+
+  /**
+   * Transform merchandise data into MerchandiseCard format
+   * grouped by gender and then by name of the product.
+   * This is used to display merchandise cards with composite data
+   * showing information about all variants of the product of that name.
+   */
+  const computeMerchandiseCards = (): Record<Gender, MerchandiseCard[]> => {
+    logger?.info('Computing merchandise cards by gender');
+    logger?.debug(`Transform <${merchandise.value.length}> items to cards`);
+
+    // group merchandise by gender
+    const merchandiseGroupedByGender = merchandise.value.reduce(
+      (acc, item) => {
+        if (!acc[item.sex as Gender]) {
+          acc[item.sex as Gender] = [];
+        }
+        acc[item.sex as Gender].push(item);
+        return acc;
+      },
+      {} as Record<Gender, Merchandise[]>,
+    );
+
+    logger?.debug(
+      `Items by gender <${JSON.stringify(merchandiseGroupedByGender, null, 2)}>`,
+    );
+
+    // for each gender, group by name and transform to MerchandiseCard
+    const cards = Object.entries(merchandiseGroupedByGender).reduce(
+      (acc, [gender, genderItems]) => {
+        logger?.debug(
+          `Processing <${genderItems.length}> items for <${gender}>`,
+        );
+
+        /**
+         * Group merchandise by name within given gender
+         * This is done to distinguish between size options for each gender.
+         */
+        const merchandiseGroupedByName = genderItems.reduce(
+          (nameAcc, item) => {
+            if (!nameAcc[item.name]) {
+              nameAcc[item.name] = [];
+            }
+            nameAcc[item.name].push(item);
+            return nameAcc;
+          },
+          {} as Record<string, Merchandise[]>,
+        );
+
+        logger?.debug(
+          `Items by name <${JSON.stringify(merchandiseGroupedByName, null, 2)}>`,
+        );
+
+        // transform each name group into MerchandiseCard
+        acc[gender as Gender] = Object.entries(merchandiseGroupedByName).map(
```

Comment:
unreadable var name `aac`, see my comment above.

Reply:
Renamed the accummulated value to `genderGroupsContainingNameGroups`.
Renamed the `merchandiseGroupedByName` variable to more precise `merchandiseWithCurrentGenderGroupedByName`